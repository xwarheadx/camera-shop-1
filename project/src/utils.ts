import { PAGE_LIMIT } from './consts';
import { reviewDataType } from '../src/types/review';

export function getPagesCount(cameraCount: number): number {
  return Math.ceil(cameraCount / PAGE_LIMIT);
}

export function getNumeric(page: string) {
  return Number(page.replace( /^\D+/g, ''));
}

export function seperatePrice(priceValue: number) {
  let returnedPrice = String(priceValue);
  if (returnedPrice.length > 3) {
    returnedPrice = `${returnedPrice.slice(0,2)} ${returnedPrice.slice(2)}`;
  }
  return returnedPrice;
}

export function validateInput(target: (EventTarget & HTMLInputElement) | (EventTarget & HTMLTextAreaElement)) {
  if (target.parentElement?.parentElement && target.name === 'rating' && (Number(target.value) > 0)) {
    if (!target.parentElement.parentElement.parentElement?.className.includes('is-valid')) {
      target.parentElement.parentElement.parentElement?.classList.add('is-valid');
      target.parentElement.parentElement.parentElement?.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name !== 'review' && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length > 0) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string'){
    if(target.value.length === 0) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length >= 5) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name === 'review' && target.value.length < 5) {
    if (!target.parentElement.parentElement.className.includes('is-invalid')) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

}
export function validateForm(data: reviewDataType) {
  let validationStatus = true;
  Object.entries(data).forEach(([key, value]) =>{
    if (key !== 'review') {
      if(value === '') {
        const failedInput = document.querySelector(`input[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
        validationStatus = false;
      }
    }
    if(key === 'review') {
      if(String(value).length < 5) {
        const failedInput = document.querySelector(`textarea[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
      }
    }
    if(key === 'rating') {
      if(value < 1) {
        const failedInput = document.querySelector(`input[name=${key}]`);
        if(failedInput && failedInput.parentElement?.parentElement){
          failedInput.parentElement.parentElement.parentElement?.classList.add('is-invalid');
        }
        validationStatus = false;
      }
    }
  });
  return validationStatus;
}


