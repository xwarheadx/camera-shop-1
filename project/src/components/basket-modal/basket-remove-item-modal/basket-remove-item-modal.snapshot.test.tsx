import { renderWithProviders } from '../../../test/utils/render-with-redux';
import BasketRemoveItemModal from './basket-remove-item-modal';

test('Render BasketRemoveItemModal', () => {
  const { asFragment } = renderWithProviders(<BasketRemoveItemModal/>);
  expect(asFragment()).toMatchSnapshot();
});
