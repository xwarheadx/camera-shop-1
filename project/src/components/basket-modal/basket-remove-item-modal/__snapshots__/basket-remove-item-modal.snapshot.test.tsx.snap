// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BasketRemoveItemModal 1`] = `
<DocumentFragment>
  <div
    class="modal is-active"
  >
    <div
      class="modal__wrapper"
    >
      <div
        class="modal__overlay"
      />
      <div
        class="modal__content"
      >
        <p
          class="title title--h4"
        >
          Удалить этот товар?
        </p>
        <div
          class="basket-item basket-item--short"
        >
          <div
            class="basket-item__img"
          >
            <picture>
              <source
                srcset="img/content/img9.webp, img/content/img9@2x.webp 2x"
                type="image/webp"
              />
              <img
                alt="Фотоаппарат «Орлёнок»"
                height="120"
                src="img/content/img9.jpg"
                srcset="img/content/img9@2x.jpg 2x"
                width="140"
              />
            </picture>
          </div>
          <div
            class="basket-item__description"
          >
            <p
              class="basket-item__title"
            >
              Фотоаппарат «Орлёнок»
            </p>
            <ul
              class="basket-item__list"
            >
              <li
                class="basket-item__list-item"
              >
                <span
                  class="basket-item__article"
                >
                  Артикул:
                </span>
                 
                <span
                  class="basket-item__number"
                >
                  O78DFGSD832
                </span>
              </li>
              <li
                class="basket-item__list-item"
              >
                Плёночная фотокамера
              </li>
              <li
                class="basket-item__list-item"
              >
                Любительский уровень
              </li>
            </ul>
          </div>
        </div>
        <div
          class="modal__buttons"
        >
          <button
            class="btn btn--purple modal__btn modal__btn--half-width"
            type="button"
          >
            Удалить
          </button>
          <a
            class="btn btn--transparent modal__btn modal__btn--half-width"
            href="#!"
          >
            Продолжить покупки
          </a>
        </div>
        <button
          aria-label="Закрыть попап"
          class="cross-btn"
          type="button"
        >
          <svg
            aria-hidden="true"
            height="10"
            width="10"
          >
            <use
              xlink:href="#icon-close"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
